### PR TITLE
Fix with Babel `react-require` plugin

### LINKED ISSUE #2: `React` is `undefined` using babel `react-require` plugin
I'm not sure if this is a Babel issue. But what's generated with Webpack 2 and Babel with the `react-require` plugin is:

```js
"use strict";
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0_react__ = __webpack_require__(/*! react */ 7);
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0_react___default = __WEBPACK_IMPORTED_MODULE_0_react__ && __WEBPACK_IMPORTED_MODULE_0_react__.__esModule ? function() { return __WEBPACK_IMPORTED_MODULE_0_react__['default'] } : function() { return __WEBPACK_IMPORTED_MODULE_0_react__; }
/* harmony import */ Object.defineProperty(__WEBPACK_IMPORTED_MODULE_0_react___default, 'a', { get: __WEBPACK_IMPORTED_MODULE_0_react___default });
module.exports = function (context) {
  return function () {
    if (!/* harmony import */__WEBPACK_IMPORTED_MODULE_0_react___default.a) {
      var React = __webpack_require__(/*! react */ 7);
    }

    return React.createElement(...);
  }.apply(context);
};
```

`__WEBPACK_IMPORTED_MODULE_0_react___default.a` is truly but `React` is not.

This issue doesn't apply if I disable the plugin.

### PR DESCRIPTION
**Pull request by faddee**
This resolves #2 by importing `React` using import statement instead if `require`-function.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,11 +16,11 @@ module.exports = function (content) {
       children = result.attributes.children,
       componentProps = renderProps(props);
 
-    let imports = '';
+    let imports = 'import React from "react";';
```
Comment (javiercf) [inline index.js]: You need to check if React is already defined before importing it, otherwise it may create some issues

Comment (faddee) [inline index.js]: You mean if it's defined globally?

Comment (faddee) [inline index.js]: Can you be more specific with which issues may occur if React is already defined?

The way I see it is that you can only define it locally by adding it as a dependency in the markdown-file.

```md
---
dependencies:
  React: react
---
```

In that case, I can do a simple check.

If it's globally defined, the import statement will just replace it locally scoped within it's own block. So I can't imagine what could go wrong. Loaders similar to this one doesn't check whether React is defined or not either. E.g: https://github.com/jhamlet/svg-react-loader/blob/master/utility/template.txt

Comment (javiercf) [inline index.js]: Ok, you are right, I was just trying it out, wanted to be sure the import would not create additional instances of React, thanks for contributing!

**Pull request merged**